RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift

```diff
@@ -303,6 +301,54 @@ struct DappInteractionFlow: Sendable, FeatureReducer {
 				}
 			)
 			return .none
+
+		case .handleInvalidPersonaOrAccounts:
+			return dismissEffect(for: state, errorKind: .invalidPersonaOrAccounts, message: nil)
+		}
+	}
+
+	private func validateAuthUsePersonaEffect(
+		requestItem: DappToWalletInteractionAuthUsePersonaRequestItem,
+		dappDefinitionAddress: DappDefinitionAddress
+	) -> Effect<Action> {
+		.run { send in
+			let identityAddress = requestItem.identityAddress
+			guard
+				let persona = try await personasClient.getPersonas()[id: identityAddress],
+				let authorizedDapp = try await authorizedDappsClient.getAuthorizedDapps()[id: dappDefinitionAddress],
+				let authorizedPersona = authorizedDapp.referencesToAuthorizedPersonas.first(where: { $0.identityAddress == identityAddress })
+			else {
+				await send(.internal(.presentPersonaNotFoundErrorAlert(reason: "")))
+				return
+			}
+
+			await send(.internal(.usePersona(requestItem, persona, authorizedDapp, authorizedPersona)))
+
+		} catch: { error, send in
+			await send(.internal(.presentPersonaNotFoundErrorAlert(reason: error.legibleLocalizedDescription)))
+		}
+	}
+
+	private func validateProofOfOwnershipRequestItemEffect(
+		requestItem: DappToWalletInteractionProofOfOwnershipRequestItem
+	) -> Effect<Action> {
+		.run { send in
+			if let identityAddress = requestItem.identityAddress {
+				// If there is an identityAddress set, verify we can access its corresponding Persona
+				let _ = try await personasClient.getPersona(id: identityAddress)
+			}
+			if let accountAddresses = requestItem.accountAddresses {
+				// For every accountAddress set, verify we can access its corresponding Account
+				let allAccounts = try await accountsClient.getAccountsOnCurrentNetwork()
+				let accounts = allAccounts.filter {
+					accountAddresses.contains($0.address)
+				}
+				if accounts.count != accountAddresses.count {
+					await send(.internal(.handleInvalidPersonaOrAccounts))
+				}
+			}
```

Comment:
When we have a `DappToWalletInteractionProofOfOwnershipRequestItem`, we will need to prove ownership of:
- 1 Persona
- N Accounts
- 1 Persona & N Accounts

In each corresponding view, we already perform the logic added here to get the actual entities given its addresses, as well as handling the case where at least one address doesn't correspond to a `Persona`/`Account` that the user owns.

However, we still want to perform this check here to avoid a situation when verifying both a persona and accounts. If the `Persona` is correctly own, but any of the `Accounts` isn't, we want to `.handleInvalidPersonaOrAccounts` before the `Persona` ownership is proven. This way we prevent the user from verifying the request _partially_, when we know the end result would be a failure.


I know, you will say that the logic to fetch the `Persona` & `Accounts` is repeated on the `PersonaProofOfOwnership` & `AccountsProofOfOwnership` reducers. Ideally, we would just update the states to inject the given entities (and not have the reducers fetch them again), but I couldn't find any way of doing so without making the code a lot uglier than how it looks now. 
Also, considering the state needs to be created before making this check (and at such point we don't have the entities yet), we would end up with either checking an optional value isn't nil or leaving the logic as it is now anyway.